### ISSUE TITLE
`TypeGuard` usages should complain about non overlapping types

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#511]
```py
def f(x: object) -> TypeGuard[int]: ...
a: str
assert f(a)  # expected error
```
